### PR TITLE
Add indentation for list and update merger

### PR DESCRIPTION
**Pull request by VCneverdie**
- Update list parser: replace tab in leading spaces with 4 spaces.
- Add `indent` for ListParagrah and ListWrapper
- Consider indent when doing post merge for list element.
> You can create a nested list by indenting one or more list items below another item.
From [Github doc Basic writing and formatting syntax](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/basic-writing-and-formatting-syntax#lists)
- Update list tests and parser tests
- Clean up stale import

### CODE REVIEW

File: src/parser/list_paragraph_parser.py [hunk 1]
```diff
@@ -6,10 +6,13 @@ def parse_ordered_list(content) -> (int, int, 'OrderedList'):
     """Parse a list paragraph, and return (begin, end, OrderedList) if parseable,
     or (-1, -1, None) that no such a element
     """
-    pattern = r'^[ ]*\d\.[ ]+(.*)$\n*'
+    pattern = r'^([ \t]*)\d\.[ ]+(.*)$\n*'
```
Comment (jinglongwu) [inline src/parser/list_paragraph_parser.py]: Thought we agreed on replace tab with 4 spaces before matching?

Comment (VCneverdie) [reply src/parser/list_paragraph_parser.py]: Addressed

Comment (jinglongwu) [reply src/parser/list_paragraph_parser.py]: Then we probably don't need to match `\t` anymore in this regex

Comment (VCneverdie) [reply src/parser/list_paragraph_parser.py]: lol you’re right

**Review by jinglongwu — APPROVED** [review]
great thought that we only replace \t in the ident part. Thinking maybe we can replace all \t into 4 spaces?

**Comment by VCneverdie** [comment]
> great thought that we only replace \t in the ident part. Thinking maybe we can replace all \t into 4 spaces?

Actually, I'm not sure if \t always equal 4 spaces. We can replace them in the indent as it doesn' really matter. (diff is large or equal to 1). I prefer not to change the \t in the content and let html or other renders render it. How do you think?

**Pull request merged**